syntax = "proto3";

option go_package = "google.golang.org/grpc/examples/helloworld/helloworld";
option java_package = "ex.grpc";
option objc_class_prefix = "HSW";

package hellostreamingworld;

// The greeting service definition.
service Greeter {
  // Sends multiple greetings
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc SayHelloReplyStream (HelloRequest) returns (stream HelloReply) {}
  rpc SayHelloRequestStream (stream HelloRequest) returns (HelloReply) {};
  rpc SayHelloStream (stream HelloRequest) returns (stream HelloReply) {};
}


// The request message containing the user's name and how many greetings
// they want.
message HelloRequest {
  string name = 1;
  string num_greetings = 2;
}

// A response message containing a greeting
message HelloReply {
  string message = 1;
}